### PR TITLE
river-filtile: init at 1.2.1

### PR DESCRIPTION
**Pull request by pkulak**
## Description of changes

Added a new layout manager for the River window manager:

https://github.com/pkulak/filtile

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Reviewed: 291198 and 291242

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by pkulak** [comment]
> Have you tried to run this package?

Yeah, I ran it with nix-build and then ran the resulting binary. Couldn't find any docs on how to do it purely with the code written, so I did it with `with import <nixpkgs> {};` instead. But at least one guide I found recommended that strategy...

**Review by IogaMaster — CHANGES_REQUESTED** [review]
## Commit standard
Commits for package initialization must follow this convention:
`pkg-name: init at <version>` so in this case your commit must be.
`river-filtile: init at 1.1.0`

## Maintainer Status
It is preferred that the packages you add have a maintainer.
Consider adding yourself?

## File structure
Consider adding the package to `pkgs/by-name/ri/river-filtile/package.nix`.
If not use the `callPackage` in the `pkgs/top-level/all-packages.nix` function to include your contribution into the nixpkgs package set.

**Comment by IogaMaster** [comment]
> > Have you tried to run this package?
> 
> Yeah, I ran it with nix-build and then ran the resulting binary. Couldn't find any docs on how to do it purely with the code written, so I did it with `with import <nixpkgs> {};` instead. But at least one guide I found recommended that strategy...

If your package is included into `pkgs/top-level/all-packages.nix` then you can run `nix run .#river-filtile`. You also need flakes enabled.

**Comment by AndersonTorres** [comment]
Typically we test a package by running

```shell
[~/Desktop]$ nix build --show-trace --print-build-logs <nixpkgs local root>#<package name>
```

where

- `<nixpkgs local root>`  is the full path of the directory of your local Nixpkgs clone (where you will find `flake.nix`) **without** the ending slash `/`;
- `<package name>` is the name you gave to the package.
  - Usually it is the same name you put on `<nixpkgs local root>/pkgs/top-level/all-packages.nix`, like:
     ```nix
        river-filtile = callPackage ../applications/misc/river-filtile/default.nix { };
     ```

    or the most  typical alternative

     ```nix
        river-filtile = callPackage ../applications/misc/river-filtile { };
     ```

Since RFC 140, this is largely simplified: instead of using `pkgs/applications/misc/river-filtile/default.nix`, you throw this same file into `pkgs/by-name/ri/river-filtile/package.nix` (and **do not touch** all-packages.nix).

**Warning**: always `git add .` before running the building command above.

**Comment by AndersonTorres** [comment]
Another style of test suite we run is `nixpkgs-review`:

```
[~/Desktop]$ nix run <nixpkgs local root>#nixpkgs-review -- pr 291392
```

The number 291392 is, precisely, the number of PR you opened.

**Comment by pkulak** [comment]
Thanks so much, guys! I have to run off, but I will have all these changes done tonight.

File: pkgs/applications/misc/river-filtile/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ lib
```
Comment (AndersonTorres) [inline pkgs/applications/misc/river-filtile/default.nix:1]: Move this file to `pkgs/by-name/ri/river-filtile/package.nix`

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by IogaMaster — CHANGES_REQUESTED** [review]

**Review by IogaMaster — COMMENTED** [review]
Other than that seems to work great!

This review has been done with [warden](https://github.com/iogamaster/warden), please report any issues!


<details><summary>Packages built</summary>
<p>

✅ river-filtile

### `nix-info -m` 
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.77, NixOS, 24.05 (Uakari), 24.05.20240216.5863c27`
 - multi-user?: `yes`
 - sandbox: `relaxed`
 - version: `nix-env (Nix) 2.19.3`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`



</p>
</details>

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -15173,6 +15173,13 @@
     githubId = 4056630;
     name = "Patrick Steinhardt";
   };
+  pkulak = {
```
[the remark this answers is not in the document]
Comment (pkulak) [reply maintainers/maintainer-list.nix:15176]: Separate commit, or separate pr?

[the remark this answers is not in the document]
Comment (IogaMaster) [reply maintainers/maintainer-list.nix:15176]: Separate commit :)

**Review by IogaMaster — APPROVED** [review]
LGTM!
Thanks for joining the nixpkgs family! We hope to see you around :wink:

**Comment by pkulak** [comment]
Thanks again for walking me through it. It'll be more easier next time. :D

**Review by AndersonTorres — APPROVED** [review]
Result of `nixpkgs-review pr 291392` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>river-filtile</li>
  </ul>
</details>

**Comment by pkulak** [comment]
Just bumped the version.

Maybe silly question, but with two approvals, does this get merged now?

**Comment by AndersonTorres** [comment]
There are no set-in-stone requirement, but yes, two approvals is a strong indication for merge.

However, we need to seek a committer. I will ping on Discourse.

Save the link below on your bookmarks!

https://discourse.nixos.org/t/prs-already-reviewed/2617

**Comment by pkulak** [comment]
Thanks!

**Comment by pinpox** [comment]
Minor nitpick: PR should be named `river-filtile: init at 1.2.1`, otherwise LGTM!

**Comment by pinpox** [comment]
Result of `nixpkgs-review pr 291392` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>river-filtile</li>
  </ul>
</details>

**Review by pinpox — APPROVED** [review]

**Pull request merged**